RevenueCatUI/Templates/V2/Components/Stack/StackComponentViewModel.swift

```diff
@@ -23,16 +23,36 @@ class StackComponentViewModel {
 
     private let component: PaywallComponent.StackComponent
     private let presentedOverrides: PresentedOverrides<PresentedStackPartial>?
+    let badgeTextViewModel: TextComponentViewModel?
 
     let viewModels: [PaywallComponentViewModel]
 
     init(
         component: PaywallComponent.StackComponent,
-        viewModels: [PaywallComponentViewModel]
+        viewModels: [PaywallComponentViewModel],
+        localizationProvider: LocalizationProvider
     ) throws {
         self.component = component
         self.viewModels = viewModels
 
+        if let badge = component.badge {
+            badgeTextViewModel = try TextComponentViewModel(
```

Comment:
Initialising the `TextComponentViewModel` of the badge in the `StackComponentViewModel` constructor so we can throw if necessary.